src/PowerShellEditorServices/Services/DebugAdapter/BreakpointSyncService.cs

```diff
@@ -0,0 +1,904 @@
+
+        return;
+    }
+
+    private void LogBreakpointError(SyncedBreakpoint breakpoint, string message)
+    {
+        // Switch to the new C# 11 string literal syntax once it makes it to stable.
+        _logger.LogError(
+            "{Message}\n" +
+            "    Server:\n" +
+            "        Id: {ServerId}\n" +
+            "        Enabled: {ServerEnabled}\n" +
+            "        Action: {Action}\n" +
+            "        Script: {Script}\n" +
+            "        Command: {Command}\n" +
+            "        Variable: {Variable}\n" +
+            "    Client:\n" +
+            "        Id: {ClientId}\n" +
+            "        Enabled: {ClientEnabled}\n" +
+            "        Condition: {Condition}\n" +
+            "        HitCondition: {HitCondition}\n" +
+            "        LogMessage: {LogMessage}\n" +
+            "        Location: {Location}\n" +
+            "        Function: {Function}\n",
+            message,
+            breakpoint.Server.Id,
+            breakpoint.Server.Enabled,
+            breakpoint.Server.Action,
+            breakpoint.Server.Script,
+            (breakpoint.Server as CommandBreakpoint)?.Command,
+            (breakpoint.Server as VariableBreakpoint)?.Variable,
+            breakpoint.Client.Id,
+            breakpoint.Client.Enabled,
+            breakpoint.Client.Condition,
+            breakpoint.Client.HitCondition,
+            breakpoint.Client.LogMessage,
+            breakpoint.Client.Location,
+            breakpoint.Client.FunctionName);
+    }
+
+    private async Task ExecuteDelegateAsync(
+            string representation,
+            ExecutionOptions? executionOptions,
+            Action<CancellationToken> action,
+            CancellationToken cancellationToken)
+    {
+        if (BreakpointApiUtils.SupportsBreakpointApis(_host.CurrentRunspace))
+        {
+            action(cancellationToken);
+            return;
+        }
+
+        await _host.ExecuteDelegateAsync(
+            representation,
+            executionOptions,
+            action,
+            cancellationToken)
+            .ConfigureAwait(false);
+    }
+
+    private static SyncedBreakpoint CreateFromServerBreakpoint(Breakpoint serverBreakpoint)
+    {
+        string? functionName = null;
+        ClientLocation? location = null;
+        if (serverBreakpoint is VariableBreakpoint vbp)
+        {
+            string mode = vbp.AccessMode switch
+            {
+                VariableAccessMode.Read => "R",
+                VariableAccessMode.Write => "W",
+                _ => "RW",
+            };
+
+            functionName = $"${vbp.Variable}!{mode}";
+        }
+        else if (serverBreakpoint is CommandBreakpoint cbp)
+        {
+            functionName = cbp.Command;
+        }
+        else if (serverBreakpoint is LineBreakpoint lbp)
+        {
+            location = new ClientLocation(
+                // TODO: fix the translation of this
```

Comment:
What's it cause right now to have this TODO?

Reply:
No idea! Hopefully I already fixed it 🙃, removed